pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -1030,6 +1035,7 @@ func TestDryRunPreemption(t *testing.T) {
 			}
 			pl := &DefaultPreemption{args: *tt.args}
 
+			rand.Seed(4)
```

Comment:
Moving this here is b/c with a non-nil podLister, `informerFactory.WaitForCacheSync(ctx.Done())` at L1023 would somehow change the internal `rand` behavior, and cause a unexpected `offset` in L1046.

Reply:
add a comment on what this is for (stable tests, I suppose)

Reply:
done.

Reply:
the remaining question (prior to this PR) is why do we need to fix the seed in the first place.

Reply:
I _think_ the original intention is to test `getOffsetAndNumCandidates()` in a deterministic manner, and 4 was chosen as the seed. https://play.golang.org/p/zkp91GhH2BG is consistent with the comment: https://github.com/kubernetes/kubernetes/blob/b049e1b9ab4a8604c0803bf4deada36116884e26/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go#L778

Reply:
nit: the comment should say: `Fixing seed to deterministically test dryRunPreemption. We need to do it after informerFactory.WaitforCacheSync which might set a seed`.

That said, this test is named TestDryRunPreemption, not `TestGetOffsetAndNumCandidates`, so it should just fix the offset rather than fixing the seed. It makes the test hard to read. @chendave can you take this?

Reply:
update. PTAL.